mpt/src/mpt.rs

```diff
@@ -0,0 +1,1647 @@
+    // sel1 and sel2 in storage leaf key: key_rlc_prev and key_rlc_mult_prev
+    // sel1 and sel2 in storage leaf value (only when leaf without branch as otherwise this info is
+    // in the branch above): whether leaf is just a placeholder
+    // sel1 and sel2 in account leaf key specifies whether nonce / balance are short / long (check
+    // nonce balance row: offset - 1)
+    sel1: Column<Advice>,
+    sel2: Column<Advice>,
+    r_table: Vec<Expression<F>>,
+    // key_rlc & key_rlc_mult used for account address, for storage key,
+    // and for mult_diff_nonce/mult_diff_balance in account_leaf_nonce_balance
+    key_rlc: Column<Advice>,
+    key_rlc_mult: Column<Advice>,
+    mult_diff: Column<Advice>,
+    keccak_table: [Column<Fixed>; KECCAK_INPUT_WIDTH + KECCAK_OUTPUT_WIDTH],
+    fixed_table: [Column<Fixed>; 3],
+    address_rlc: Column<Advice>, /* The same in all rows of a modification. The same as
+                                  * address_rlc computed in the account leaf key row. Needed to
+                                  * enable lookup for storage key/value (to have address RLC in
+                                  * the same row as storage key/value). */
+    counter: Column<Advice>, 
+}
+
+#[derive(Clone, Copy, Debug)]
+pub enum FixedTableTag {
+    RMult,
+    Range16,
+    Range256,
+    RangeKeyLen256,
+}
+
+struct ProofVariables<F> {
+    modified_node: u8,
+    s_mod_node_hash_rlc: F,
+    c_mod_node_hash_rlc: F,
+    node_index: u8,
+    acc_s: F,
+    acc_mult_s: F,
+    acc_account_s: F,
+    acc_mult_account_s: F,
+    acc_account_c: F,
+    acc_mult_account_c: F,
+    acc_nonce_balance_s: F,
+    acc_mult_nonce_balance_s: F,
+    acc_nonce_balance_c: F,
+    acc_mult_nonce_balance_c: F,
+    acc_c: F,
+    acc_mult_c: F,
+    key_rlc: F, /* used first for account address, then for storage key */
+    key_rlc_mult: F,
+    extension_node_rlc: F,
+    key_rlc_prev: F, /* for leaf after placeholder extension/branch, we need to go one level
+                      * back to get previous key_rlc */
+    key_rlc_mult_prev: F,
+    mult_diff: F,
+    key_rlc_sel: bool, /* If true, nibble is multiplied by 16, otherwise by 1. */
+    is_branch_s_placeholder: bool,
+    is_branch_c_placeholder: bool,
+    drifted_pos: u8, /* needed when leaf turned into branch and leaf moves into a branch where
+                      * it's at drifted_pos position */
+    rlp_len_rem_s: i32, /* branch RLP length remainder, in each branch children row this value
+                         * is subtracted by the number of RLP bytes in
+                         * this row (1 or 33) */
+    rlp_len_rem_c: i32,
+    is_extension_node: bool,
+    is_even: bool,
+    is_odd: bool,
+    is_short: bool,
+    is_long: bool,
+    rlc1: F,
+    rlc2: F,
+    nonce_value_s: F,
+    balance_value_s: F,
+    before_account_leaf: bool,
+}
+
+impl<F: Field> ProofVariables<F> {
+    fn new() -> ProofVariables<F> {
```

Comment:
There are some options we don't have to declare all fields here. We can derive `Default` trait for ProofVariables but in some of the fields, we want them to be `F::one()` or `true`.

One option is we keep the `new` method but override the fields. The downside is that we now have `new` and `default` methods, which can be confusing because people don't know which one is the preferred initialization.
```
impl<F: Field> ProofVariables<F> {
    fn new() -> Self {
        Self {
            key_rlc_mult: F::one(),
            key_rlc_mult_prev: F::one(),
            mult_diff: F::one(),
            key_rlc_sel: true,
            before_account_leaf: true,
            ..Default::default()
        }
    }
}
```
Option 2 is to use `derivative` crate https://stackoverflow.com/a/69240986/2754799. The benefit is the `default` method initialized the desired fields. The downside is that we introduced a new crate for this relatively trivial case.

Reply:
Agreed, better this way, the first option used in [2cd8d15](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/560/commits/2cd8d15e5ad2c244db8c9761545f7f850e198269).